src/core/counter/mod.rs

```diff
@@ -10,55 +10,48 @@ pub struct Counter {
 }
 
 impl Counter {
-    pub fn new_with_code_and_count_b64(code: &str, count_b64: &str) -> Result<Counter> {
-        let count = if count_b64.is_empty() { 1_u32 } else { util::b64_to_u32(count_b64)? };
-
-        Counter::new_with_code_and_count(code, count)
-    }
-
-    pub fn new_with_code_and_count(code: &str, count: u32) -> Result<Counter> {
-        if code.is_empty() {
-            return err!(Error::EmptyMaterial("empty code".to_string()));
-        }
-
-        let szg = tables::sizage(code)?;
-        let cs = szg.hs + szg.ss;
-        if szg.fs != cs || cs % 4 != 0 {
-            // unreachable
-            // code validated and unless sizages are broken this cannot be reached
-            return err!(Error::InvalidCodeSize(format!(
-                "whole code size not a multiple of 4: cs = {cs}, fs = {}",
-                szg.fs
-            )));
-        }
+    pub fn new(
+        count: Option<u32>,
+        count_b64: Option<&str>,
+        code: Option<&str>,
+        qb64b: Option<&mut Vec<u8>>,
+        qb64: Option<&str>,
+        qb2: Option<&mut Vec<u8>>,
+        strip: Option<bool>,
+    ) -> Result<Self> {
+        let strip = strip.unwrap_or(false);
+
+        if let Some(code) = code {
+            let count = if let Some(count) = count {
+                count
+            } else if let Some(count_b64) = count_b64 {
+                util::b64_to_u32(count_b64)?
+            } else {
+                1
+            };
 
-        if count > 64_u32.pow(szg.ss) - 1 {
-            return err!(Error::InvalidVarIndex(format!(
-                "invalid count for code: count = {count}, code = '{code}'"
-            )));
+            Self::new_with_code_and_count(code, count)
+        } else if let Some(qb64b) = qb64b {
+            let counter = Self::new_with_qb64b(qb64b)?;
+            if strip {
+                let szg = tables::sizage(&counter.code())?;
+                let length = szg.fs as usize;
+                qb64b.resize(length, b'\x00');
```

Comment:
I suspect  `strip=true` should return the rest of the bytes (which were not parsed), but according to the documentation `resize` will rather return the beginning of the vector
```
let mut vec = vec![1, 2, 3, 4];
vec.resize(2, 0);
assert_eq!(vec, [1, 2]);
```

Reply:
This is a good catch, I may have done this backwards. I questioned the utility of removing the rest of the stream, I probably just read the python code wrong. I'll check.

Reply:
What this may mean is that this isn't tested in the python implementation, since I copied the tests.

Reply:
Yeah you're totally correct, thanks for this Artem

Reply:
I made #100

Reply:
Actually the tests were copied before I made this change, now that I'm thinking more. I likely just made some to match implementation during the add.

Reply:
Anyhow I'm fixing this in the Prefixer PR.